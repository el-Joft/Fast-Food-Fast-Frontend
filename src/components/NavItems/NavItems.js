import React from 'react';
import { Link } from 'react-router-dom';
import './NavItems.scss';

const NavItem = () => {
  const items = [
    {
      text: 'Home',
      link: '/',
    },
    {
      text: 'About Us',
      link: '/about',
    },
    {
      text: 'Login',
      link: '/login',
    },
    {
      text: 'Sign Up',
      link: '/signup',
    },
  ];

  const showItems = () => items.map((item, i) => (
    <li key={i}>
      <Link to={item.link}>
        {item.text}
      </Link>
    </li>
  ));

  return (
    <div>
      <nav>
        <ul className="nav-bar">
          {showItems()}
          <li className="cart-section">
            <a href="checkout.html">
              <img src="/public/images/icons/cart.png" alt="" />
              <span>
Cart(
                <span id="cartSystem" />
)
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default NavItem;
