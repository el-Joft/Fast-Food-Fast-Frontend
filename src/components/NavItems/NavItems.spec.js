import React from 'react';
import { shallow } from 'enzyme';
import { NavItem } from './NavItems';


describe('shallow testing', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
      },
    };
  });
  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NavItem {...props} />);
    expect(wrapper.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NavItem {...props} />);
    const li = wrapper.find('li');
    expect(li.length).toEqual(5);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NavItem {...props} />);
    const li = wrapper.find('Link');
    expect(li.length).toEqual(5);
  });
  // it('should test a method inside the component', () => {
  //   const logoutClick = jest.fn();
  //   wrapper = shallow(<NavItem {...props} />);
  //   wrapper.find('Link').simulate('click');
  //   wrapper.instance();
  //   expect(logoutClick).toHaveBeenCalled();
  // });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<NavItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
