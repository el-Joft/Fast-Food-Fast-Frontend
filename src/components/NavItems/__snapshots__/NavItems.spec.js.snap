// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow testing should test the component if it renders anything 1`] = `
<div>
  <nav>
    <ul
      className="nav-bar"
    >
      <li
        key="0"
      >
        <Link
          replace={false}
          to="/menu"
        >
          Menu
        </Link>
      </li>
      <li
        key="1"
      >
        <Link
          replace={false}
          to="/about"
        >
          About Us
        </Link>
      </li>
      <li
        key="2"
      >
        <Link
          replace={false}
          to="/dashboard"
        >
          Welcome
        </Link>
      </li>
      <li>
        <Link
          onClick={[Function]}
          replace={false}
          to="/logout"
        >
          Logout
        </Link>
      </li>
      <li
        className="cart-section"
      >
        <Link
          replace={false}
          to="/checkout"
        >
          <img
            alt=""
            src={Object {}}
          />
          <span>
            Cart(
            0
            )
          </span>
        </Link>
      </li>
    </ul>
  </nav>
</div>
`;
